Modules/_elementtree.c

```diff
@@ -3894,6 +3894,34 @@ _elementtree_XMLParser_close_impl(XMLParserObject *self)
     }
 }
 
+/*[clinic input]
+_elementtree.XMLParser.flush
+
+[clinic start generated code]*/
+
+static PyObject *
+_elementtree_XMLParser_flush_impl(XMLParserObject *self)
+/*[clinic end generated code: output=42fdb8795ca24509 input=effbecdb28715949]*/
+{
+    if (!_check_xmlparser(self)) {
+        return NULL;
+    }
+
+    elementtreestate * const st = self->state;
+
+    if (EXPAT(st, SetReparseDeferralEnabled) == NULL) {
+        return NULL;
+    }
+
+    EXPAT(self->state, SetReparseDeferralEnabled)(self->parser, XML_FALSE);
+
+    PyObject* const res = expat_parse(st, self, "", 0, XML_FALSE);
```

Comment:
`const` is redundant.

Reply:
@serhiy-storchaka that's an unexpected ask since adding `const` is a feature in my world — something that Rust understood and made immutability default. Is there some document where CPython asks to omit `const` wherever it does not strictly affect const correctness as with our case here? If we were in a pull request in libexpat with our roles switched I would ask to insert `const` here during review from you, not dropping it. I can drop it, but I would rather not. What do you think?

Reply:
The style of the new code should be consistent with the style of the old code, this is the main reason. In this case it does not have any effect except increasing the length of the line and attracting an excessive attention with its uncommon style; both harms readability.

There may be code guides which require `const` whether it is possible, and we should follow the code guide. But AFAIK excessive `const`s considered bad practice in C++ in last 20 or 30 years. Note the difference between `X const * p` and `X * const p` -- the former is meaningful, while the later is meaningless for local variables, unless you follow a specific restrictive code guide.

Reply:
> The style of the new code should be consistent with the style of the old code, this is the main reason. In this case it does not have any effect except increasing the length of the line and attracting an excessive attention with its uncommon style; both harms readability.
>
> There may be code guides which require `const` whether it is possible, and we should follow the code guide. But AFAIK excessive `const`s considered bad practice in C++ in last 20 or 30 years.

@serhiy-storchaka I can accept that as your opinion, and will drop the two `const`s that you wanted to remove with the next push in a minute.

> Note the difference between `X const * p` and `X * const p` -- the former is meaningful, while the later is meaningless for local variables, unless you follow a specific restrictive code guide.

It's not meaningless, but the difference is clear. (I've been teaching C const correctness to students 13 years ago in university, [related slides in German](https://github.com/hartwork/c-kurs-pointers) are still up if curious :smiley:)

Reply:
Pushed, 2x `const` gone.